src/core.js

```diff
@@ -45,18 +45,111 @@ function assert(condition, message) {
   }
 }
 
+/**
+ * @new
+ *
+ * AnyMatch can be used inside mapping keys to indicate a wildcard match.
+ * Most commonly used within CALL effect argument lists:
+ *   [call(fn, AnyMatch), result]
+ *
+ * If AnyMatch appears in the middle of an array, it matches a single element at that spot (and continues).
+ */
+const AnyMatch = Symbol.for('@@redux-saga-test-engine/ANY_MATCH')
+
+function isPlainObject(o) {
+  return Object.prototype.toString.call(o) === '[object Object]'
+}
+
+// Matches arrays where `keyArr` may contain AnyMatch.
+// - If AnyMatch is the **last** element, it matches the rest of `actualArr` (zero or more items).
+// - If AnyMatch appears **not last**, it matches exactly one item at that position.
+function arrayMatchesWithAny(actualArr, keyArr) {
+  let i = 0 // index into keyArr
+  let j = 0 // index into actualArr
+
+  while (i < keyArr.length && j < actualArr.length) {
+    const kItem = keyArr[i]
+
+    if (kItem === AnyMatch) {
+      const restWildcard = i === keyArr.length - 1
+      if (restWildcard) {
+        // AnyMatch at the end means "match the rest (including zero elements)"
+        return true
+      }
```

Comment:
Minor: I like guard clauses to be one-liners when possible
```suggestion
      // AnyMatch at the end means "match the rest (including zero elements)"
      if (restWildcard) return true
```